interfaces/builtin/nvidia_tegra_drivers_support.go

```diff
@@ -0,0 +1,175 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const nvidiaTegraDriversSupportSummary = `allows hardware access to NVIDIA tegra platforms`
+
+const nvidiaTegraDriversSupportBaseDeclarationSlots = `
+  nvidia-tegra-drivers-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const nvidiaTegraDriversSupportConnectedPlugAppArmor = `
+# This is inverse of
+# https://forum.snapcraft.io/t/call-for-testing-chromium-62-0-3202-62/2569/46
+# As nvidia-assemble snap needs to create the static & dynamic MAJOR
+# chrdevs for all other snaps to have access to. Specifically
+/{,usr/}bin/mknod ixr,
+allow capability mknod,
+
+# To read dynamically allocated MAJOR for nvidia-uvm
+@{PROC}/modules r,
+@{PROC}/devices r,
+@{PROC}/driver/nvidia/params r,
+@{PROC}/driver/nvidia/capabilities/mig/monitor r,
+@{PROC}/driver/nvidia/capabilities/mig/config r,
+@{PROC}/sys/vm/mmap_min_addr r,
+
+/sys/devices/soc0/platform r,
+/sys/devices/soc0/soc_id r,
+/sys/devices/soc0/revision r,
+/sys/devices/soc0/major r,
+/sys/devices/platform/bus@0/3810000.fuse/fuse/nvmem r,
+
+/dev/nvmap rw,
+/dev/dri/renderD128 rw,
+/dev/nvgpu/igpu0/power rw,
+/dev/nvgpu/igpu0/ctrl rw,
+/dev/nvgpu/igpu0/prof rw,
+/dev/char/498:1 rw,
```

Comment:
Can you document twhat the major:minor numbers are here please?

Reply:
IIRC this major falls within the dynamic major number allocation range. Is this hardcoded by some tegra driver? How is the number allocated by the driver, is it hardcoded, or rather requested by the driver?

Reply:
a few lines above there's:
```
# To read dynamically allocated MAJOR for nvidia-uvm
```
suggesting that indeed the major number can change? If anything, there would be an AppArmor rule granting rw to /dev/char (making this a really privileged interface?), and then we'd use udev & seccomp to selectively give device access